packages/cli/src/config/extension.test.ts

```diff
@@ -1401,3 +1548,143 @@ describe('enableExtension', () => {
     expect(activeExtensions[0].name).toBe('ext1');
   });
 });
+
+describe('extension', () => {
+  describe('checkForExtensionUpdate', () => {
```

Comment:
I say this partly because merge conflicts are going to be a real pain, but also: WDYT of pulling the update logic out into the config/extensions/ directory into its own file?